app/components/views/Offset/index.tsx

```diff
@@ -170,6 +172,11 @@ export const Offset = (props: Props) => {
 
   // effects
   useEffect(() => {
+    // round the value for the edge case where a user inputs a fractional value
+    // when fiat isn't selected to pay. Example (100.5 -> 101)
+    if (paymentMethod === "fiat" && !Number.isInteger(parseFloat(quantity))) {
+      setQuantity(Math.round(Number(quantity)).toString());
+    }
```

Comment:
What's the thinking behind this bit here?

Thinking out loud, is this to cover cases where a fractional input value could be:
- copy pasted into the input
- when a user switches payment methods
- guard/validate against url params

Reply:
> when a user switches payment methods

Yep as above:

-> A user wishes to pay in USDC so they select USDC from the payment method dropdown 
-> They enter 100.55 in the quantity input, they change their mind and want to pay by credit card.
-> Because by default we don't reset the input field if the payment method is changed, the value is now 100.55 in the input field. Because nothing changed in the input field, an onChange/onKeyDown event is never triggered so the validation doesn't kick in to prevent decimal values.
-> There is a number of ways I guess this could be handled, for example -> clear the input field completely, round the value (which is what I am doing here) or we could attempt to somehow trigger an onChange event on the input field to trigger the validation.